qiskit_experiments/library/characterization/cr_hamiltonian.py

```diff
@@ -0,0 +1,472 @@
+                ),
+                pulse.control_channels(*self.physical_qubits)[0],
+            )
+            # add cancellation tone
+            if not np.isclose(opt.amp_t, 0.0):
+                pulse.play(
+                    pulse.GaussianSquare(
+                        duration,
+                        amp=opt.amp_t,
+                        sigma=sigma,
+                        risefall_sigma_ratio=opt.risefall,
+                    ),
+                    pulse.drive_channel(self.physical_qubits[1]),
+                )
+            else:
+                pulse.delay(duration, pulse.drive_channel(self.physical_qubits[1]))
+
+            # place holder for empty drive channels. this is necessary due to known pulse gate bug.
+            pulse.delay(duration, pulse.drive_channel(self.physical_qubits[0]))
+
+        cr_circuit.add_calibration(
+            gate=cr_gate,
+            qubits=self.physical_qubits,
+            schedule=cross_resonance,
+            params=[duration],
+        )
+
+        return cr_circuit
+
+    def _net_duration(
+        self,
+        duration: float,
+        sigma: float,
+    ) -> float:
+        """Calculate net cross resonance pulse duration considering the pulse edges.
+
+        Args:
+            duration: Parameter object representing a duration of cross resonance pulse.
+            sigma: Sigma of Gaussian edges.
+
+        Returns:
+            Net duration in units of sec.
+        """
+        opt = self.experiment_options
+
+        flat_top_width = duration - 2 * opt.risefall * sigma
+        net_edge_width = np.sqrt(2 * np.pi * sigma ** 2)
+
+        return flat_top_width + net_edge_width
+
+    def circuits(self, backend: Optional[Backend] = None) -> List[QuantumCircuit]:
+
+        opt = self.experiment_options
+        prefactor = 1.0
+        timing_constraints = getattr(backend.configuration(), "timing_constraints", dict())
+        granularity = timing_constraints.get("granularity", 1)
+
+        try:
+            dt_factor = backend.configuration().dt
+        except AttributeError as ex:
+            raise AttributeError("Backend configuration does not provide time resolution.") from ex
+
+        if opt.unit != "dt":
+            if opt.unit != "s":
+                prefactor *= apply_prefix(1.0, opt.unit)
+            prefactor /= dt_factor
+
+        # Define pulse gate of cross resonance tone
+        duration = circuit.Parameter("duration")
+        sigma_dt = prefactor * opt.sigma
+
+        template_circuits = list()
+        for control_state in (0, 1):
+            for meas_basis in ("x", "y", "z"):
+                tomo_circ = QuantumCircuit()
+                tomo_circ.add_register(2, 1)
+
+                # state prep
+                if control_state:
+                    tomo_circ.x(0)
+
+                # add cross resonance
+                tomo_circ.compose(
+                    other=self._define_decomposition(backend, duration, sigma_dt),
```

Comment:
Can we make a better split between the schedule and the circuit? I.e. do 
```
        cr_gate = circuit.Gate("cr_tone", num_qubits=2, params=[duration])

        tomo_circ.append(cr_gate, [0, 1])
```
in `circuits` instead of doing it in  `_define_decomposition`?

Reply:
The motivation of decomp method (that returns a circuit) is to allow one to reuse custom gate in the instmap/backend. If we define `cr_gate`, we always need to provide full schedule of `cr_gate`. For example, ECR uses x gate. The pulse def can be automatically provided if we define the cross resonance as a circuit.

Reply:
That's fine. What I'm looking for is a separation between the schedule logic and the circuit logic. I.e. I would define:
```python
def _cr_gate() -> QuantumCircuit:
    circ = QuantumCircuit(2)
    circ.append(cr_gate, [0, 1])
    return circ
```
for the single-tone and
```python
def _cr_gate() -> QuantumCircuit:
    cr_circuit = QuantumCircuit(2)
    cr_circuit.append(cr_gate, [0, 1])
    cr_circuit.x(0)
    cr_circuit.rz(np.pi, 1)
    cr_circuit.append(cr_gate, [0, 1])
    cr_circuit.rz(-np.pi, 1)
    return cr_circuit
```
and have a separate function to bring in the schedule.

Reply:
I also thought about this. The problem is if we have more than two types of CR gates, it is hard to define schedules in a separate method (we need to use dictionary to communicate). 

I guess you meant this enables to provide schedule as an experiment option, which enhances the flexibility of choice of envelopes, however, users always need to write schedules, and single liner execution is no longer possible. If we have user option for entire schedule, parameter based construction, etc... again the logic becomes complicated.